src/components/smart_house/smart_house.ts

```diff
@@ -0,0 +1,59 @@
+import { DeviceInterface } from '../device/types';
+
+export default class SmartHouse {
+    protected name: string;
+    protected devicesList: Map<number, DeviceInterface>;
+
+    constructor(name: string) {
+        this.name = name;
+        this.devicesList = new Map();
+    }
+
+    public getName(): string {
+        return this.name;
+    }
+    public setName(name: string): void {
+        if (name.length > 0) {
+            this.name = name;
+        }
+    }
+
+    private generateId(): number {
+        return Math.floor(Math.random() * 100000000 + 1);
+    }
+
+    public addDevice(device: DeviceInterface): void {
```

Comment:
consider using more meaningful naming prior to DeviceInterface. 
Because in a such scenario it even would be better to use IDevice :)

Reply:
Did I understand you properly, that it would be better to remane `DeviceInterface `to `BaseDeviceInterface`?

Reply:
No to smth like DeviceOperations or more descriptive